src/topology/running.rs

```diff
@@ -463,169 +463,175 @@ impl RunningTopology {
     /// Starts new and changed pieces of topology.
     pub(crate) fn spawn_diff(&mut self, diff: &ConfigDiff, mut new_pieces: Pieces) {
         // Sources
-        for name in &diff.sources.to_change {
-            info!(message = "Rebuilding source.", name = ?name);
-            self.spawn_source(name, &mut new_pieces);
+        for id in &diff.sources.to_change {
+            info!(message = "Rebuilding source.", id = ?id);
+            self.spawn_source(id, &mut new_pieces);
         }
 
-        for name in &diff.sources.to_add {
-            info!(message = "Starting source.", name = ?name);
-            self.spawn_source(name, &mut new_pieces);
+        for id in &diff.sources.to_add {
+            info!(message = "Starting source.", id = ?id);
+            self.spawn_source(id, &mut new_pieces);
         }
 
         // Transforms
-        for name in &diff.transforms.to_change {
-            info!(message = "Rebuilding transform.", name = ?name);
-            self.spawn_transform(name, &mut new_pieces);
+        for id in &diff.transforms.to_change {
+            info!(message = "Rebuilding transform.", id = ?id);
+            self.spawn_transform(id, &mut new_pieces);
         }
 
-        for name in &diff.transforms.to_add {
-            info!(message = "Starting transform.", name = ?name);
-            self.spawn_transform(name, &mut new_pieces);
+        for id in &diff.transforms.to_add {
+            info!(message = "Starting transform.", id = ?id);
+            self.spawn_transform(id, &mut new_pieces);
         }
 
         // Sinks
-        for name in &diff.sinks.to_change {
-            info!(message = "Rebuilding sink.", name = ?name);
-            self.spawn_sink(name, &mut new_pieces);
+        for id in &diff.sinks.to_change {
+            info!(message = "Rebuilding sink.", id = ?id);
+            self.spawn_sink(id, &mut new_pieces);
         }
 
-        for name in &diff.sinks.to_add {
-            info!(message = "Starting sink.", name = ?name);
-            self.spawn_sink(name, &mut new_pieces);
+        for id in &diff.sinks.to_add {
+            info!(message = "Starting sink.", id = ?id);
+            self.spawn_sink(id, &mut new_pieces);
         }
     }
 
-    fn spawn_sink(&mut self, name: &str, new_pieces: &mut builder::Pieces) {
-        let task = new_pieces.tasks.remove(name).unwrap();
+    fn spawn_sink(&mut self, id: &str, new_pieces: &mut builder::Pieces) {
+        let task = new_pieces.tasks.remove(id).unwrap();
         let span = error_span!(
             "sink",
+            component_id = %task.id(),
             component_kind = "sink",
-            component_name = %task.name(),
             component_type = %task.typetag(),
+            // maintained for compatibility
+            component_name = %component_id,
```

Comment:
Looks like this should be:
```suggestion
            component_name = %task.id,
```

Reply:
Yes indeed, thanks!

Reply:
6345ff8